### PR TITLE
update toc schema

### PR DESCRIPTION
**Pull request by herohua**

### CODE REVIEW

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[yih-markdown-test/schemas/toc.schema.json](https://github.com/herohua/markdown-test/blob/update-toc-schema/yih-markdown-test/schemas/toc.schema.json) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstoragesandbox2.blob.core.windows.net/report/2018%5C8%5C4%5C55506563-8dd2-b513-6276-829687a360b3%5CPullRequest%5C201808040716456416-29%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=i1BbI8jUgeWrEWK%2B2gG%2FEipyZ3jFeeDDF51uXaHyVoo%3D&st=2018-08-04T07%3A13%3A09Z&se=2018-09-04T07%3A18%3A09Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[yih-markdown-test/schemas/toc.schema.json](https://github.com/herohua/markdown-test/blob/update-toc-schema/yih-markdown-test/schemas/toc.schema.json) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstoragesandbox2.blob.core.windows.net/report/2018%5C8%5C4%5C55506563-8dd2-b513-6276-829687a360b3%5CPullRequest%5C201808040754352467-29%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=9PPwJsTQ3YDTU%2FcDW3TDIqg3VtLvKB6Oa%2FhMV%2BCyARs%3D&st=2018-08-04T07%3A50%3A49Z&se=2018-09-04T07%3A55%3A49Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by superyyrrzz** [comment]
> 		"name": {

name should be required, and not empty

---
Refers to: yih-markdown-test/schemas/toc.schema.json:20 in a9b8b7e. [](commit_id = a9b8b7ecf51ca68e354d236dc6326b8a4a77529d, deletion_comment = False)

File: yih-markdown-test/schemas/toc.schema.json [hunk 1]
```diff
@@ -27,14 +27,24 @@
...
 			"contentType": "href"
 		},
 		"uid": {
 			"type": "string",
 			"description": "The uid the TOC node navigates to. Optional because nodes can exist just to parent other nodes.",
 			"contentType": "uid"
 		},
+		"topicHref": {
+			"type": "string",
+			"description": "[TODO]",
```
Comment (OsmondJiang) [inline yih-markdown-test/schemas/toc.schema.json]: >[TODO] [](start = 19, length = 6)

Specifies the topic href of the TOC Item. It is useful when href is linking to a folder or linking to a toc.

File: yih-markdown-test/schemas/toc.schema.json [hunk 2]
```diff
@@ -27,14 +27,24 @@
...
+			"contentType": "href"
+		},
+		"tocHref": {
+			"type": "string",
+			"description": "[TODO]",
```
Comment (OsmondJiang) [inline yih-markdown-test/schemas/toc.schema.json]: >[TODO] [](start = 19, length = 6)

Specifies the toc href/folder of the TOC Item. It is useful when href is linking to topic file.

**Comment by OsmondJiang** [comment]
> 			"properties": {

add tocHref and topicHref here?

---
Refers to: yih-markdown-test/schemas/toc.schema.json:57 in a9b8b7e. [](commit_id = a9b8b7ecf51ca68e354d236dc6326b8a4a77529d, deletion_comment = False)

File: yih-markdown-test/schemas/toc.schema.json [hunk 3]
```diff
@@ -27,14 +27,24 @@
...
+			"contentType": "href"
+		},
```
Comment (superyyrrzz) [inline yih-markdown-test/schemas/toc.schema.json]: Mention that this is only needed in breadcrumb?

File: yih-markdown-test/schemas/toc.schema.json [hunk 4]
```diff
@@ -71,6 +75,17 @@
 					"minItems": 1
 				}
 			},
+			"not": {
```
Comment (live1206) [inline yih-markdown-test/schemas/toc.schema.json]: >not [](start = 4, length = 3)

Should this be "oneof"? "An instance validates successfully against this keyword if it validates successfully against exactly one schema defined by this keyword's value"

Comment (herohua) [reply yih-markdown-test/schemas/toc.schema.json]: this is to forbid the existence of both "uid" and "href"

---
In reply to: [207769389](https://github.com/herohua/markdown-test/pull/29#discussion_r207769389) [](ancestors = 207769389)

**Comment by herohua** [comment]
> 		"name": {

will add `minLength`

---
In reply to: [410565601](https://github.com/herohua/markdown-test/pull/29#issuecomment-410565601) [](ancestors = 410565601)

---
Refers to: yih-markdown-test/schemas/toc.schema.json:20 in a9b8b7e. [](commit_id = a9b8b7ecf51ca68e354d236dc6326b8a4a77529d, deletion_comment = False)

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[yih-markdown-test/schemas/toc.schema.json](https://github.com/herohua/markdown-test/blob/update-toc-schema/yih-markdown-test/schemas/toc.schema.json) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstoragesandbox2.blob.core.windows.net/report/2018%5C8%5C6%5C55506563-8dd2-b513-6276-829687a360b3%5CPullRequest%5C201808060611027445-29%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=PGNCN%2B97yFrdGiq20r32WlcQB6SNcfwcTA4kfh1k5Ko%3D&st=2018-08-06T06%3A07%3A26Z&se=2018-09-06T06%3A12%3A26Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[yih-markdown-test/schemas/toc.schema.json](https://github.com/herohua/markdown-test/blob/update-toc-schema/yih-markdown-test/schemas/toc.schema.json) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstoragesandbox2.blob.core.windows.net/report/2018%5C8%5C6%5C55506563-8dd2-b513-6276-829687a360b3%5CPullRequest%5C201808060632074632-29%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=bdcYDhUcRBWHstbfY94vVMPO2J8sPXcNqmiR49WjyD4%3D&st=2018-08-06T06%3A30%3A18Z&se=2018-09-06T06%3A35%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[yih-markdown-test/schemas/toc.schema.json](https://github.com/herohua/markdown-test/blob/update-toc-schema/yih-markdown-test/schemas/toc.schema.json) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstoragesandbox2.blob.core.windows.net/report/2018%5C8%5C6%5C55506563-8dd2-b513-6276-829687a360b3%5CPullRequest%5C201808060634534025-29%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=jIUElMVpqLRwN%2FYILZLsME0JTnshbdBkLG8pqqalj7U%3D&st=2018-08-06T06%3A31%3A40Z&se=2018-09-06T06%3A36%3A40Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: yih-markdown-test/schemas/toc.schema.json (hunk 3 above)
Comment (herohua) [reply yih-markdown-test/schemas/toc.schema.json]: don't want to introduce the concept of breadcrumb in toc schema, also in v3 we actually support `tocHref` with the meaning outlined here

---
In reply to: [207768567](https://github.com/herohua/markdown-test/pull/29#discussion_r207768567) [](ancestors = 207768567)

File: yih-markdown-test/schemas/toc.schema.json [hunk 5]
```diff
@@ -71,6 +96,22 @@
...
+						"required": [ "href", "uid" ]
+					},
+					{
+						"required": [ "uid", "topicHref" ]
+					},
+					{
+						"required": [ "uid", "tocHref" ]
+					},
+					{
+						"required": [ "href", "tocHref" ]
```
Comment (OsmondJiang) [inline yih-markdown-test/schemas/toc.schema.json]: >"required": [ "href", "tocHref" ] [](start = 6, length = 33)

they are supported

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[yih-markdown-test/schemas/toc.schema.json](https://github.com/herohua/markdown-test/blob/update-toc-schema/yih-markdown-test/schemas/toc.schema.json) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstoragesandbox2.blob.core.windows.net/report/2018%5C8%5C6%5C55506563-8dd2-b513-6276-829687a360b3%5CPullRequest%5C201808060702265002-29%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=uNd4UM9pphM2bwjymF2z5FphcH9o%2Bb3yAou2OdhWaps%3D&st=2018-08-06T06%3A58%3A42Z&se=2018-09-06T07%3A03%3A42Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by superyyrrzz** [comment]
> 	"$schema": "https://dotnet.github.io/docfx/schemas/v1.0/schema.json#",

The current logic of toc cannot be fully described by DocFX schema 1.0. What about use pure JSON schema?

---
Refers to: yih-markdown-test/schemas/toc.schema.json:2 in d171a62. [](commit_id = d171a6261ca157c8755a7ca12730c8a7c0ba8da6, deletion_comment = False)

**Comment by herohua** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[yih-markdown-test/schemas/toc.schema.json](https://github.com/herohua/markdown-test/blob/update-toc-schema/yih-markdown-test/schemas/toc.schema.json) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstoragesandbox2.blob.core.windows.net/report/2018%5C8%5C7%5C55506563-8dd2-b513-6276-829687a360b3%5CPullRequest%5C201808070916456860-29%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=GWJ4dAgY%2FEGCXB7dYJynxHeKfTX2tnDEEovEg5z%2BeTs%3D&st=2018-08-07T09%3A13%3A27Z&se=2018-09-07T09%3A18%3A27Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**